packages/playground/windows/playground/MainPage.cpp

```diff
@@ -28,6 +28,16 @@ MainPage::MainPage() {
     x_rootComponentNameCombo().IsEditable(true);
     x_entryPointCombo().IsEditable(true);
   }
+
+#if !defined(USE_HERMES)
```

Comment:
Having a preprocessor definition outside of the MSRN boundary will create some issues when we're on binary distribution, since the definitions used to build Microsoft.ReactNative.DLL won't always match with this. Could you add an API internal to the boundary to query whether a JS engine is available instead?

Reply:
I think that would just look like adding a function to IReactInstanceSettings and its IDL which use the definition there, instead of here.

Reply:
IReactInstanceSettings is an input into the instance.  I don't think it makes sense as a place to expose details about how the dll was built. It seems like something that could be done as a separate API altogether if we really need to expose these things externally. -- I'm also not sure if we want to be exposing such an API in the first place.

Reply:
Would the alternative be "show everything in test UI and let things noop if they weren't built?" IIRC we will fallback to Chakra if we request an unavailable JS engine instead of failing fast, but maybe that behavior can be changed.

Reply:
Ideally, once we have ABI-safe JSI, we can just pass the "engine" as a parameter to whatever API we have for instance creation.

---
In reply to: [467330017](https://github.com/microsoft/react-native-windows/pull/5666#discussion_r467330017) [](ancestors = 467330017)

Reply:
We could do that, since the property needs to be separately defined anyway (it looks like React.Cpp.props is no longer included in Playground). It will assert, but that's ignorable.

---
In reply to: [467330698](https://github.com/microsoft/react-native-windows/pull/5666#discussion_r467330698) [](ancestors = 467330698)

Reply:
If we assert, leaving everything available as an option seems okay to me then, just as long as we get clear feedback when choosing something that is unavailable.

React.cpp.props is only included internal to the MSRN boundary. ReactUwp didn't really have that clear separation, but for MSRN we let users template projects be a bit less opinionated, and don't want to let code slip in where they use raw React Native outside the boundary.